plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/AbstractTestIcebergConnectorTest.java

```diff
@@ -1786,4 +1813,47 @@ public void testIncorrectIcebergFileSizes()
 
         dropTable("test_iceberg_file_size");
     }
+
+    @Override
+    protected TestTable createTableWithDefaultColumns()
+    {
+        throw new SkipException("Iceberg connector does not support column default values");
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup dataMappingTestSetup)
+    {
+        String typeName = dataMappingTestSetup.getTrinoTypeName();
+        if (typeName.equals("tinyint")
+                || typeName.equals("smallint")
+                || typeName.startsWith("char(")) {
+            // These types are not supported by Iceberg
+            return Optional.of(dataMappingTestSetup.asUnsupported());
+        }
+
+        // According to Iceberg specification all time and timestamp values are stored with microsecond precision.
+        if (typeName.equals("time")) {
+            return Optional.of(new DataMappingTestSetup("time(6)", "TIME '15:03:00'", "TIME '23:59:59.999999'"));
```

Comment:
What happens for `new DataMappingTestSetup("time", "TIME '15:03:00'", "TIME '23:59:59.999'")` case?
do we get an unsupported type during CTAS?

we should probably have a test case for time(3) and one for time(6) (and 9, 12)
and we would return `asUnsupported` for all except time(6)

-- can be follow up

Reply:
I agree. I'll make a follow-up adding those types.